src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.IncrementalMemberEditAnalyzer.cs

```diff
@@ -232,169 +216,6 @@ async Task ExecuteAnalyzersAsync(
-                DiagnosticAnalyzer analyzer,
-                DocumentAnalysisExecutor executor,
-                SyntaxNode changedMember,
-                int changedMemberId,
-                ImmutableArray<TextSpan> oldMemberSpans,
-                Func<DiagnosticAnalyzer, DocumentAnalysisExecutor, CancellationToken, Task<ImmutableArray<DiagnosticData>>> computeAnalyzerDiagnosticsAsync,
-                CancellationToken cancellationToken)
-            {
-                // We are performing semantic span-based analysis for member-only edit scenario.
-                // Instead of computing the analyzer diagnostics for the entire document,
-                // we have computed the new diagnostics just for the edited member span.
-                Debug.Assert(executor.AnalysisScope.Span.HasValue);
-                Debug.Assert(executor.AnalysisScope.Span.Value == changedMember.FullSpan);
-
-                // We now try to get the new document diagnostics by performing an incremental update:
-                //   1. Re-using all the old cached diagnostics outside the edited member node from a prior
-                //      document snapshot, but with updated diagnostic spans.
-                //      AND
-                //   2. Replacing old diagnostics for the edited member node in a prior document snapshot
-                //      with the new diagnostics for this member node in the latest document snaphot.
-                // If we are unable to perform this incremental diagnostics update,
-                // we fallback to computing the diagnostics for the entire document.
-                var tree = changedMember.SyntaxTree;
-                var text = tree.GetText(cancellationToken);
-                if (TryGetUpdatedDocumentDiagnostics(existingData, oldMemberSpans, diagnostics, tree, text, changedMember, changedMemberId, out var updatedDiagnostics))
-                {
-#if DEBUG_INCREMENTAL_ANALYSIS
-                    await ValidateMemberDiagnosticsAsync(executor, analyzer, updatedDiagnostics, cancellationToken).ConfigureAwait(false);
-#endif
-                    return updatedDiagnostics;
-                }
-                else
-                {
-                    // Incremental diagnostics update failed.
-                    // Fallback to computing the diagnostics for the entire document.
-                    var documentExecutor = executor.With(executor.AnalysisScope.WithSpan(null));
-                    return await computeAnalyzerDiagnosticsAsync(analyzer, documentExecutor, cancellationToken).ConfigureAwait(false);
-                }
-
-#if DEBUG_INCREMENTAL_ANALYSIS
-                static async Task ValidateMemberDiagnosticsAsync(DocumentAnalysisExecutor executor, DiagnosticAnalyzer analyzer, ImmutableArray<DiagnosticData> diagnostics, CancellationToken cancellationToken)
-                {
-                    executor = executor.With(executor.AnalysisScope.WithSpan(null));
```

Comment:
we would end up falling into this codepath.  but that's the codepath we already call *above* the call to GetUpdatedDiagnosticsForMemberEditAsync